### PR TITLE
Global error handler cleanup - UpDown Counter, Historgram, Gauge.

### PR DESCRIPTION
**Pull request by lalitb**
Fixes #
Design discussion issue (if applicable) #

## Changes

Please provide a brief description of the changes here.

## Merge requirement checklist

* [ ] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-rust/blob/main/CONTRIBUTING.md) guidelines followed
* [ ] Unit tests added/updated (if applicable)
* [ ] Appropriate `CHANGELOG.md` files updated for non-trivial, user-facing changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: opentelemetry-sdk/src/metrics/meter.rs [hunk 1]
```diff
@@ -296,14 +296,20 @@ impl SdkMeter {
     {
         let validation_result = validate_instrument_config(builder.name.as_ref(), &builder.unit);
         if let Err(err) = validation_result {
-            global::handle_error(err);
+            otel_error!(
+                name: "InstrumentCreationFailed",
+                meter_name = self.scope.name.as_ref(),
+                instrument_name = builder.name.as_ref(),
+                message = "Measurements from this updown counter will be ignored.",
```
Comment (utpilla) [inline opentelemetry-sdk/src/metrics/meter.rs:303]:
```suggestion
                message = "Measurements from this UpDownCounter will be ignored.",
```

File: opentelemetry-sdk/src/metrics/meter.rs [hunk 2]
```diff
@@ -328,14 +340,20 @@ impl SdkMeter {
     {
         let validation_result = validate_instrument_config(builder.name.as_ref(), &builder.unit);
         if let Err(err) = validation_result {
-            global::handle_error(err);
+            otel_error!(
+                name: "InstrumentCreationFailed",
+                meter_name = self.scope.name.as_ref(),
+                instrument_name = builder.name.as_ref(),
+                message = "Measurements from this gauge will be ignored.",
```
Comment (utpilla) [inline opentelemetry-sdk/src/metrics/meter.rs:347]: Let's use the capital letter for the first letter of the instrument name.

```suggestion
                message = "Measurements from this Gauge will be ignored.",
```

**Review by utpilla — APPROVED** [review]
Let a suggestion about using capital letter for the first letter of each instrument name. That's how we use it in the spec docs as well.

If we go this route, this will need to be updated for the previously merged PRs as well.

**Review by cijothomas — APPROVED** [review]

**Comment by lalitb** [comment]
> Let a suggestion about using capital letter for the first letter of each instrument name. That's how we use it in the spec docs as well.
> 
> If we go this route, this will need to be updated for the previously merged PRs as well.

Good point. Have updated for all the instruments.

**Pull request merged**